mono/mini/interp/interp-internals.h

```diff
@@ -145,9 +141,57 @@ struct _InterpFrame {
 	/* exception info */
 	const unsigned short  *ip;
 	MonoException     *ex;
+
+	// These are all pointers, but assumed to be within
+	// 2GB of the frame, i.e. all stack allocated, and the frame is too.
+	// This is a bit painful to debug, granted.
+	//
+	// As the pointers can be beyond 4GB, NULL has a special representation
+	// of 0. The values cannot coincide with frame itself, so that is not
+	// ambiguous.
+	//
+	// These are at the end of InterpFrame to sort by size, to minimize padding for alignment.
+	//
+	gint32 parent_offset;     // parent
+	gint32 retval_offset;     // parent
+	gint32 stack_args_offset; // parent
```

Comment:
Probably `stack_args` should be put back due to frequent use, or interpreter should cache the full pointer in a local.
The others I think are rare enough.

Also this could use and ifdef 64bit since it only costs on 32bit, buys nothing. If we care about 32bit, 32bit interpreter.

Also you could quibble over how many bits these should be. Default Linux stack being 8MB, like 23 bits, and alignment, but 32 is convenient and a reduction.